FLUID-6148: Initial pass at docs for new potentia II features (#136)
fluid-project/infusion-docs: src/documents/APIChangesFrom3_0To4_0.md

---
title: API Changes from 3.0 to 4.0
category: Infusion
---

This page contains a list of the features, APIs, and etc. that have changed in Infusion 4.0.

## Framework Changes

### Core Framework Changes

#### DataSources

The DataSource implementation was refactored around two pseudoevents `onRead` and `onWrite` as described
in the [DataSource API](DataSourceAPI.md).

> An browser implementation of <code>fluid.dataSource.url</code> has been provided, based on the XMLHttpRequest API.

Reviewer: ```suggestion
A browser implementation of `fluid.dataSource.url` has been provided, based on the [XMLHttpRequest API](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest).
```
